BraveRewardsUI/Settings/Tips/Details/TipsDetailViewController.swift

```diff
@@ -264,7 +264,11 @@ extension TipsDetailViewController {
   fileprivate func getTipsThisMonth() -> BalanceReportInfo {
     let month = Date().currentMonthNumber
     let year = Date().currentYear
-    return state.ledger.balanceReport(for: ActivityMonth(rawValue: month) ?? .any, year: Int32(year))
+    var report = BalanceReportInfo()
+    state.ledger.balanceReport(for: ActivityMonth(rawValue: month) ?? .any, year: Int32(year)) {
+      if let balance = $0 { report = balance }
```

Comment:
This function is async.. but it returns the report before the function completes?
Is it truly non-async but has a callback instead of return-statement?

Reply:
It is non-escaping, executed serially. They just changed the API on native libs side for memory reasons I believe?